pkg/envoy/server.go

```diff
@@ -174,164 +174,141 @@ func StartXDSServer(stateDir string) *XDSServer {
+
+	hcmConfig := &envoy_config_http.HttpConnectionManager{
+		StatPrefix: "proxy",
+		HttpFilters: []*envoy_config_http.HttpFilter{{
+			Name: "cilium.l7policy",
+			ConfigType: &envoy_config_http.HttpFilter_TypedConfig{
+				TypedConfig: toAny(&cilium.L7Policy{
+					AccessLogPath:  s.accessLogPath,
+					Denied_403Body: denied403body,
+				}),
 			},
-		},
-		Transparent: &wrappers.BoolValue{Value: true},
-		SocketOptions: []*envoy_api_v2_core.SocketOption{{
-			Description: "Listener socket mark",
-			Level:       syscall.SOL_SOCKET,
-			Name:        syscall.SO_MARK,
-			Value:       &envoy_api_v2_core.SocketOption_IntValue{IntValue: 0xB00}, // egress
-			State:       envoy_api_v2_core.SocketOption_STATE_PREBIND,
+		}, {
+			Name: "envoy.router",
 		}},
-		// FilterChains: []*envoy_api_v2_listener.FilterChain
-		ListenerFilters: []*envoy_api_v2_listener.ListenerFilter{{
-			Name: "cilium.bpf_metadata",
-			ConfigType: &envoy_api_v2_listener.ListenerFilter_Config{
-				Config: &structpb.Struct{Fields: map[string]*structpb.Value{
-					"is_ingress":                      {Kind: &structpb.Value_BoolValue{BoolValue: false}},
-					"may_use_original_source_address": {Kind: &structpb.Value_BoolValue{BoolValue: false}},
-					"bpf_root":                        {Kind: &structpb.Value_StringValue{StringValue: bpf.GetMapRoot()}},
+		StreamIdleTimeout: &duration.Duration{}, // 0 == disabled
+		RouteSpecifier: &envoy_config_http.HttpConnectionManager_RouteConfig{
+			RouteConfig: &envoy_api_v2.RouteConfiguration{
+				VirtualHosts: []*envoy_api_v2_route.VirtualHost{{
+					Name:    "default_route",
+					Domains: []string{"*"},
+					Routes: []*envoy_api_v2_route.Route{{
+						Match: &envoy_api_v2_route.RouteMatch{
+							PathSpecifier: &envoy_api_v2_route.RouteMatch_Prefix{Prefix: "/"},
+							Grpc:          &envoy_api_v2_route.RouteMatch_GrpcRouteMatchOptions{},
+						},
+						Action: &envoy_api_v2_route.Route_Route{
+							Route: &envoy_api_v2_route.RouteAction{
+								ClusterSpecifier: &envoy_api_v2_route.RouteAction_Cluster{
+									Cluster: clusterName,
+								},
+								Timeout:        &duration.Duration{Seconds: requestTimeout},
+								MaxGrpcTimeout: &duration.Duration{Seconds: maxGRPCTimeout},
+								RetryPolicy: &envoy_api_v2_route.RetryPolicy{
+									RetryOn:       "5xx",
+									NumRetries:    &wrappers.UInt32Value{Value: numRetries},
+									PerTryTimeout: &duration.Duration{Seconds: retryTimeout},
+								},
+							},
+						},
+					}, {
+						Match: &envoy_api_v2_route.RouteMatch{
+							PathSpecifier: &envoy_api_v2_route.RouteMatch_Prefix{Prefix: "/"},
+						},
+						Action: &envoy_api_v2_route.Route_Route{
+							Route: &envoy_api_v2_route.RouteAction{
+								ClusterSpecifier: &envoy_api_v2_route.RouteAction_Cluster{
+									Cluster: clusterName,
+								},
+								Timeout: &duration.Duration{Seconds: requestTimeout},
+								//IdleTimeout: &duration.Duration{Seconds: idleTimeout},
```

Comment:
It might be worth noting here that this is set below but I was able to notice it pretty quickly so it's probably fine anyway.